pkg/controller/v1alpha1/inferencegraph/knative_reconciler.go

```diff
@@ -175,12 +175,24 @@ func createKnativeService(componentMeta metav1.ObjectMeta, graph *v1alpha1api.In
 	}
 
 	// Only adding this env variable "PROPAGATE_HEADERS" if router's headers config has the key "propagate"
-	value, exists := config.Headers["propagate"]
-	if exists {
+	configHeaders, configHeadersExists := config.Headers["propagate"]
+	specHeaders, specHeadersExists := graph.Spec.Headers["propagate"]
+
+	// Use spec headers over router config headers
+	if specHeadersExists {
+		log.Info("Found spec header")
```

Comment:
Can we log the headers here ? 
Like log.Info("Found spec header", "specHeaders", specHeaders)